### DISCUSSION TITLE
Is a Context Provider good practice?

### DISCUSSION THREAD

**Discussion by MateusAmin** [rrousselGit/riverpod#1386] [General]
This makes it easy to access the context in a Riverpod app. I can create a snackbar message (I see people just using snackbar providers instead), interface with the router, and have a [async proof](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html) way of accessing the current context.

For example:

```dart
/// A context provider to be able to grab an up to date context from ref.
final contextProvider = StateProvider<BuildContext>((ref) {
  return StatelessElement(const Text('as'));
});

Future<void> main() async {
runApp(
      ProviderScope(
        child: MaterialApp.router(
          builder: (context, child) {
            return Consumer(
              builder: (context, ref, _) {
                ref.read(contextProvider.notifier).state = context;

                return child!;
              },
            );
          },
          routeInformationParser: enumRouter.routeInformationParser,
          routerDelegate: enumRouter.routerDelegate,
        ),
      ),
    ),
  );
}
```

Just include the builder in your widget tests.

**Comment by rrousselGit** [comment]
No this is a bad idea. Don't pass a BuildContext to your providers

**Reply by MateusAmin** [reply]
Thanks for the response! I hope open source is treating you well.

I saw this as an example of a pattern I could do instead:

https://github.com/rrousselGit/riverpod/discussions/1331#discussioncomment-2509825
